name.abuchen.portfolio/src/name/abuchen/portfolio/model/ClientFactory.java

```diff
@@ -1572,6 +1575,34 @@ private static void fixNullSecurityProperties(Client client)
         }
     }
 
+    private static void fixInvestmentPlanTypes(Client client)
+    {
+        for (InvestmentPlan plan : client.getPlans())
+        {
+            InvestmentPlan.Type type;
+
+            if (plan.getPortfolio() != null)
+            {
+                if (plan.getSecurity() == null)
+                    throw new IllegalArgumentException("security is null with a set portfolio for" + plan.getName()); //$NON-NLS-1$
+
+                type = InvestmentPlan.Type.BUY_OR_DELIVERY;
+            }
+            else
+            {
+                if (plan.getAccount() == null)
+                    throw new IllegalArgumentException("portfolio and account are not set for " + plan.getName()); //$NON-NLS-1$
+
+                if (plan.getSecurity() == null)
+                    type = (plan.getAmount() >= 0) ? InvestmentPlan.Type.DEPOSIT : InvestmentPlan.Type.REMOVAL;
+                else
+                    throw new IllegalArgumentException("security is set with a set account for " + plan.getName()); //$NON-NLS-1$
```

Comment:
I removed the exceptions here. The problem: if the exception is thrown, then the file is never opened and the user has no chance to "fix" the entries.

Previously, those checks were done when generating the transactions. Then the user could fix the plan, for example deleting the plan and re-creating one, or editing the plan. All error are anyway programming errors because they should never happen if the editing logic is correct.